src/components/runbooks/RunbookForm.js

```diff
@@ -0,0 +1,202 @@
+import React, {Component} from 'react';
+import {observable, action} from 'mobx';
+import {observer} from 'mobx-react';
+import {clone} from 'lodash';
+import cx from 'classnames';
+
+import FileInput from '../FileInput';
+import DownloadFileButton from '../DownloadFileButton';
+
+@observer
+export default class RunbookForm extends Component {
+  @observable key = Date.now()
+
+  changeParameter = (index, attr, value) => {
+    let parameters = clone(this.props.data.parameters.slice());
+    parameters[index][attr] = value;
+    this.props.onParametersChange(parameters);
+  }
+
+  addParameter = () => {
+    let parameters = clone(this.props.data.parameters.slice());
+    parameters.push({name: '', default: '', type: 'string'});
+    this.props.onParametersChange(parameters);
+  }
+
+  @action
+  removeParameter = (index) => {
+    let parameters = clone(this.props.data.parameters.slice());
+    parameters.splice(index, 1);
+    this.props.onParametersChange(parameters);
+    this.key = Date.now();
+  }
+
+  changeTag = (index, value) => {
+    let tags = clone(this.props.data.tags.slice());
+    tags[index] = value;
+    this.props.onTagsChange(tags);
+  }
+
+  addTag = () => {
+    let tags = clone(this.props.data.tags.slice());
+    tags.push('');
+    this.props.onTagsChange(tags);
+  }
+
+  @action
+  removeTag = (index) => {
+    let tags = clone(this.props.data.tags.slice());
+    tags.splice(index, 1);
+    this.props.onTagsChange(tags);
+    this.key = Date.now();
+  }
+
+  render() {
+    const {data, errors, onChange, onRunbookUpload} = this.props;
+    return (
+      <div className='runbook-form' key={this.key}>
+        <div className={cx('form-group', {'has-error': !!errors.name})}>
+          <label htmlFor='runbook-name' className='control-label'>{'Name'}</label>
+          <input
+            type='text'
+            className='form-control'
+            id='runbook-name'
+            name='name'
+            defaultValue={data.name}
+            onChange={onChange}
+          />
+          <p className='help-block'>{errors.name}</p>
+        </div>
+        <div className='runbook-tags'>
+          <label className='control-label'>{'Tags'}</label>
+          {data.tags.map((tag, index) => {
+            const error = (errors.tags || [])[index];
+            return (
+              <div className='row' key={index}>
```

Comment:
Can we use tag itself here? Using index for key is last resort option and here there are better options